.github/workflows/ci.yml

```diff
@@ -0,0 +1,181 @@
+name: Run all tests
+
+on: [push, pull_request]
+
+jobs:
+  setup:
+    name: CI test
+    env:
+      IGNORE_PATHS: tests/fixtures
+    runs-on: 'ubuntu-latest'
+
+    services:
+      postgres:
+        image: postgres:9.6
+        env:
+          POSTGRES_USER: 'postgres'
+          POSTGRES_HOST_AUTH_METHOD: 'trust'
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 3
+        ports:
+          - 5432:5432
+
+      mariadb:
+        image: mariadb:10.5
+        env:
+          MYSQL_USER: 'root'
+          MYSQL_ALLOW_EMPTY_PASSWORD: "true"
+        ports:
+          - 3306:3306
+        options: >-
+          --health-cmd="mysqladmin ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 3
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - php: '7.1'
+            moodle-branch: 'MOODLE_34_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.2'
+            moodle-branch: 'MOODLE_34_STABLE'
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.1'
+            moodle-branch: 'MOODLE_35_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.2'
+            moodle-branch: 'MOODLE_35_STABLE'
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.1'
+            moodle-branch: 'MOODLE_36_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.2'
+            moodle-branch: 'MOODLE_36_STABLE'
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.1'
+            moodle-branch: 'MOODLE_37_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.2'
+            moodle-branch: 'MOODLE_37_STABLE'
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_38_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_38_STABLE'
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_39_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_39_STABLE'
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_310_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_310_STABLE'
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_311_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_311_STABLE'
+            database: 'pgsql'
+            node: '14.15'
```

Comment:
this matrix is quite hard to follow and will grow out of control when more versions are added...maybe consider simplifying that to something like this: https://github.com/catalyst/moodle-auth_saml2/blob/MOODLE_39_STABLE/.github/workflows/39-master.yml#L43-L55

Reply:
Hey @jwalits - updated with the given reference. Please review.. Thank you...